File: year2020/month12/day1207/ms_1207.cs
using System;

namespace algorithm_116
{
    class ms_1207
    {
        static bool solution(string s)
        {
            bool isNumber = true;
            bool isInRange = true;
            int count = 0;

            foreach(char token in s)
            {
                int temp = token - '0';

                if(!(0 <= temp && temp <= 9))
                {
                    isNumber = false;
                }

                count++;
            }

            if(!(count == 4 || count == 6))
            {
                isInRange = false;
            }

            return isNumber && isInRange;
        } 

        static void Main(string[] args)
        {
            String s = Console.ReadLine();

            if(1 <= s.Length && s.Length <= 9)
            {
                Console.WriteLine(solution(s));
            }
        }
    }
}

File: year2020/month12/day1207/ms_1207.cpp
#include <iostream>
#include <string>

using namespace std;

bool solution(string s);

int main() {
	string s;

	getline(cin, s);

	if (1 <= s.length() && s.length() <= 8) {
		cout << solution(s) << endl;
	}
	return 0;
}

bool solution(string s) {
	bool isNumber = true;
	bool isInRange = true;
	int i = 0;

	while (s[i] != NULL) {
		if (!(0 <= s[i]-'0' && s[i]-'0' <= 9)) {
			isNumber = false;
		}

		i++;
	}

	if (i != 4 && i != 6) {
		isInRange = false;
	}

	return isNumber && isInRange;
}